patches/zos-v3.18.0.patch

```diff
@@ -150,6 +150,34 @@ index 40668a3e6b..f04506aad2 100644
  # else /* unknown architecture */
  #  define ARCHITECTURE_ID ""
  # endif
+diff --git a/Modules/CheckTypeSize.cmake b/Modules/CheckTypeSize.cmake
+index 2b07b7cc03..036f913aed 100644
+--- a/Modules/CheckTypeSize.cmake
++++ b/Modules/CheckTypeSize.cmake
+@@ -117,6 +117,12 @@ function(__check_type_size_impl type var map builtin language)
+   endif()
+   set(bin ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/CheckTypeSize/${var}.bin)
+   configure_file(${__check_type_size_dir}/CheckTypeSize.c.in ${src} @ONLY)
++  if(CMAKE_SYSTEM_NAME MATCHES "OS390")
++    set(CMAKE_C_FLAGS_SAVE ${CMAKE_C_FLAGS})
++    set(CMAKE_CXX_FLAGS_SAVE ${CMAKE_CXX_FLAGS})
++    string(REGEX REPLACE "-(qASCII|qascii|fzos-le-char-mode=ascii);?" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
++    string(REGEX REPLACE "-(qASCII|qascii|fzos-le-char-mode=ascii);?" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```

Comment:
Thanks for raising the PR!

Is the regex replace case insensitive? Would this work on -qAscii? There's also -Wc,ASCII which is accepted by xlclang.

An alternative would be to append -qNOASCII and the negative option of -fzos-le-char-mode=ascii to CMAKE_*FLAGS

Reply:
Hi @IgorTodorovskiIBM 

Thanks!  I made the change based on your review comment now.  It's now based on the no-ASCII logic  in `CMakeDetermineCompilerABI.cmake` (instead of `CMakeDetermineCompilerId.cmake`).